### PR TITLE
Add documentation examples of consider-using-f-string

### PR DESCRIPTION
**Pull request by timmartin**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Adds documentation examples for the `consider-using-f-string` checker. There didn't seem to be any examples in the `pylint-errors` repo that I could see, so I wrote a new one.

Ref #5953

### CODE REVIEW

File: doc/data/messages/c/consider-using-f-string/bad.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+menu = ('eggs', 'spam')
+
+order = "%s and %s" % menu # [consider-using-f-string]
+
+new_order = "{1} and {0}".format(menu[0], menu[1]) # [consider-using-f-string]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/consider-using-f-string/bad.py:5]:
```suggestion
old_order = "%s and %s: %.2f ¤" % menu # [consider-using-f-string]
beginner_order = menu[0] + "and " + menu[1] + ": " + str(menu[2]) + " ¤"
joined_order = "and ".join(menu) 
format_order = "{} and {}: {:0.2f} ¤".format(menu[0], menu[1], menu[2]) # [consider-using-f-string]
named_format_order = "{eggs} and {spam}: {price:0.2f} ¤".format(eggs=menu[0], spam=menu[1], price=menu[2]) # [consider-using-f-string]
template_order = Template('$eggs and $spam: $price ¤').substitute(eggs=menu[0], spam=menu[1], price=menu[2])
```

File: doc/data/messages/c/consider-using-f-string/bad.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+menu = ('eggs', 'spam')
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/consider-using-f-string/bad.py:1]:
```suggestion
from string import Template

menu = ('eggs', 'spam', 42.42)
```

File: doc/data/messages/c/consider-using-f-string/good.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+menu = ('eggs', 'spam')
+
+order = f"{menu[0]} and {menu[1]}"
+
+new_order = f"{menu[1]} and {menu[0]}"
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/c/consider-using-f-string/good.py:5]:
```suggestion
f_string_order = f"{menu[0]} and {menu[1]}: {menu[2]:0.2f} ¤"
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I think we need a detail.rst saying f-string performance at a lot better with a link to this tweet from a python core dev : https://twitter.com/raymondh/status/1205969258800275456. Also added more technical f-string with float formatting because often it's not known that this kind of thing can be done with fstring too

File: doc/data/messages/c/consider-using-f-string/bad.py (hunk 1 above)
Comment (timmartin) [reply doc/data/messages/c/consider-using-f-string/bad.py:5]: Hmm, thanks for introducing me to the `¤` sign!

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you @timmartin !

**Pull request merged**